hail/src/main/scala/is/hail/types/physical/stypes/interfaces/SNDArray.scala

```diff
@@ -626,6 +627,63 @@ object SNDArray {
       work.firstDataAddress, lwork.toI))
     cb.ifx(info.cne(0), cb._fatal(s"LAPACK error DGEQRF. Error code = ", info.toS))
   }
+
+  def syevr_query(cb: EmitCodeBuilder, jobz: String, uplo: String, n: Value[Int], region: Value[Region]): (SizeValue, SizeValue) = {
+    val WorkAddress = cb.memoize(region.allocate(8L, 8L))
+    val IWorkAddress = cb.memoize(region.allocate(4L, 4L))
+    val info = cb.memoize(Code.invokeScalaObject19[String, String, String, Int, Long, Int, Double, Double, Int, Int, Double, Long, Long, Int, Long, Long, Int, Long, Int, Int](LAPACK.getClass, "dsyevr",
+      jobz, "A", uplo,
+      n, 0, n,
+      0, 0, 0, 0,
+      0,
+      0, 0, n,
+      0,
+      WorkAddress, -1,
+      IWorkAddress, -1))
+    cb.ifx(info.cne(0), cb._fatal(s"LAPACK error DSYEVR. Failed size query. Error code = ", info.toS))
+    val LWork = cb.memoize(Region.loadDouble(WorkAddress).toL)
+    val LIWork = cb.memoize(Region.loadInt(IWorkAddress).toL)
+    (SizeValueDyn(cb.memoize((LWork > 0).mux(LWork, 1))), SizeValueDyn(cb.memoize((LIWork > 0).mux(LIWork, 1))))
+  }
+
+  def syevr(cb: EmitCodeBuilder, uplo: String, A: SNDArrayValue, W: SNDArrayValue, Z: Option[(SNDArrayValue, SNDArrayValue)], Work: SNDArrayValue, IWork: SNDArrayValue): Unit = {
+    assertMatrix(A)
+    assertColMajor(cb, "orgqr", A)
+    assertVector(W, Work, IWork)
+    assert(IWork.pt.elementType.virtualType == TInt32)
+
+    val n = A.shapes(0)
+    A.assertHasShape(cb, Array(n, n), "syevr: A must be square")
+    W.assertHasShape(cb, Array(n), "syevr: W has wrong size")
```

Comment:
Is there a way to include what the shapes it got were? ie "expected square matrix, got 12-by-11".

Reply:
I fixed `assertHasShape` to print the shapes.